### ISSUE TITLE
Detect cfg based on an undeclared feature

### ISSUE THREAD

**Issue by gendx** [rust-lang/rust-clippy#12634]
### What it does

Detect and warn on uses of `cfg(feature = "foo")` where the feature ("foo") isn't declared in the Cargo.toml file.

### Advantage

The cfg always evaluates to false, which is likely not intended (e.g. typo, feature that was renamed in the Cargo.toml file but not everywhere in the code, etc.).

### Drawbacks

I don't see any reason to have a `cfg` based on a feature that doesn't exist.

### Example

```toml
# Cargo.toml
[features]
bar = []
```

```rust
let x = 0;
// Feature "foo" isn't defined in the Cargo.toml file.
#[cfg(feature = "foo")]
x = 42;
```

**Comment by y21** [comment]
Closing as a duplicate of #11649. This already exists unstably in cargo as `-Zcheck-cfg`, and is close to being stabilized it looks like (tracking issue: rust-lang/rust#82450)
```
warning: unexpected `cfg` condition value: `foo`
 --> src/main.rs:4:11
  |
4 |     #[cfg(feature = "foo")]
  |           ^^^^^^^^^^^^^^^ help: remove the condition
  |
  = note: no expected values for `feature`
  = help: consider adding `foo` as a feature in `Cargo.toml`
```

Thanks for opening this regardless!
